Source/WebCore/style/Styleable.cpp

```diff
@@ -672,6 +673,24 @@ void Styleable::updateCSSTransitions(const RenderStyle& currentStyle, const Rend
                 continue;
             updateCSSTransitionsForStyleableAndProperty(*this, property, currentStyle, newStyle, generationTime);
         }
+
+        HashSet<AtomString> animatableCustomProperties;
+        auto gatherAnimatableCustomPropertiesInMap = [&](const CustomPropertyValueMap& customPropertyValueMap) {
+            for (auto [customProperty, customPropertyValue] : customPropertyValueMap) {
+                if (!customPropertyValue)
+                    continue;
+                auto& variantValue = customPropertyValue->value();
+                if (std::holds_alternative<CSSCustomPropertyValue::SyntaxValue>(variantValue)
+                    || std::holds_alternative<CSSCustomPropertyValue::SyntaxValueList>(variantValue))
+                    animatableCustomProperties.add(customProperty);
+            }
+        };
+        gatherAnimatableCustomPropertiesInMap(currentStyle.inheritedCustomProperties());
+        gatherAnimatableCustomPropertiesInMap(currentStyle.nonInheritedCustomProperties());
+        gatherAnimatableCustomPropertiesInMap(newStyle.inheritedCustomProperties());
+        gatherAnimatableCustomPropertiesInMap(newStyle.nonInheritedCustomProperties());
+        for (auto& customProperty : animatableCustomProperties)
+            updateCSSTransitionsForStyleableAndProperty(*this, customProperty, currentStyle, newStyle, generationTime);
```

Comment:
you should test with `RenderStyle::customPropertiesEqual` before doing all this work.

Reply:
This is already being done in `updateCSSTransitionsForStyleableAndProperty()` and we don't do this for regular `CSSPropertyID` above.